src/Spice86.Shared/Emulator/Memory/SegmentedAddress.cs

```diff
@@ -34,17 +17,40 @@ public SegmentedAddress(ValueTuple<ushort, ushort> segmentOffset) : this(segment
     public SegmentedAddress(ushort segment, ushort offset) {
         Segment = segment;
         Offset = offset;
+        Linear = (uint)(Segment << 4) + Offset;
     }
 
+    /// <summary>
+    /// Gets the offset value of the address.
+    /// </summary>
+    public ushort Offset { get; }
+
+    /// <summary>
+    /// Gets the segment value of the address.
+    /// </summary>
+    public ushort Segment { get; }
+
+    /// <summary>
+    /// Gets the linear value of the address.
+    /// </summary>
+    public uint Linear { get; }
+
     /// <summary>
     /// Compares the current SegmentedAddress object with another SegmentedAddress object.
     /// </summary>
     /// <param name="other">The SegmentedAddress to compare with the current SegmentedAddress.</param>
     /// <returns>A value that indicates the relative order of the objects being compared.</returns>
     public int CompareTo(SegmentedAddress other) {
-        uint x = ToPhysical();
-        uint y = other.ToPhysical();
-        return x < y ? -1 : x == y ? 0 : 1;
+        return this < other ? -1 : this == other ? 0 : 1;
+    }
+
+    /// <summary>
+    /// Determines whether the current SegmentedAddress object is equivalent to another SegmentedAddress object.
+    /// </summary>
+    /// <param name="other">The object to compare with the current SegmentedAddress object.</param>
+    /// <returns>true if the objects are equivalent; otherwise, false.</returns>
+    public bool Equals(SegmentedAddress other) {
+        return Linear == other.Linear;
```

Comment:
This will break CfgCpu on some corner cases.
We store a map of SegmentedAddress -> Instruction and we want to have a different entry if the instruction is accessed via a different segment, because ultimately for some, this changes the way the instruction behaves.